src/workerd/api/node/tests/timers-global-override-test.js

```diff
@@ -0,0 +1,386 @@
+export const testGlobalSetIntervalReturnsObject = {
+  async test() {
+    const interval = globalThis.setInterval(() => {}, 1000);
+
+    // Verify it's an object, not a number
+    strictEqual(
+      typeof interval,
+      'object',
+      'setInterval should return an object'
+    );
+
+    // Verify it has the expected methods
+    strictEqual(
+      typeof interval.refresh,
+      'function',
+      'Interval should have refresh method'
+    );
+    strictEqual(
+      typeof interval.ref,
+      'function',
+      'Interval should have ref method'
+    );
+    strictEqual(
+      typeof interval.unref,
+      'function',
+      'Interval should have unref method'
+    );
+    strictEqual(
+      typeof interval.hasRef,
+      'function',
+      'Interval should have hasRef method'
+    );
+
+    // Verify Symbol.toPrimitive allows using as numeric ID for backward compat
+    strictEqual(
+      typeof +interval,
+      'number',
+      'Interval should be convertible to number'
+    );
+    ok(+interval > 0, 'Interval numeric ID should be positive');
+
+    // Clean up
+    globalThis.clearInterval(interval);
+  },
+};
+
+// Test that clearTimeout accepts both Timeout objects and numeric IDs
+export const testClearTimeoutAcceptsBothTypes = {
+  async test() {
+    // Test clearing with Timeout object
+    const t1 = globalThis.setTimeout(() => {
+      throw new Error('timeout1 should have been cleared');
+    }, 100);
+    globalThis.clearTimeout(t1);
+
+    // Test clearing with numeric ID (backward compatibility via Symbol.toPrimitive)
+    const t2 = globalThis.setTimeout(() => {
+      throw new Error('timeout2 should have been cleared');
+    }, 100);
+    globalThis.clearTimeout(+t2);
+
+    // Wait to ensure cleared timeouts don't fire
+    await new Promise((r) => globalThis.setTimeout(r, 200));
+  },
+};
+
+// Test that clearInterval accepts both Timeout objects and numeric IDs
+export const testClearIntervalAcceptsBothTypes = {
+  async test() {
+    // Test clearing with Timeout object
+    const i1 = globalThis.setInterval(() => {
+      throw new Error('interval1 should have been cleared');
+    }, 50);
+    globalThis.clearInterval(i1);
+
+    // Test clearing with numeric ID (backward compatibility via Symbol.toPrimitive)
+    const i2 = globalThis.setInterval(() => {
+      throw new Error('interval2 should have been cleared');
+    }, 50);
+    globalThis.clearInterval(+i2);
+
+    // Wait to ensure cleared intervals don't fire
+    await new Promise((r) => globalThis.setTimeout(r, 200));
+  },
+};
+
+// Test that the refresh() method works correctly
+export const testTimeoutRefresh = {
+  async test() {
+    let callCount = 0;
+
+    // Create a timeout that would fire in 50ms
+    const timeout = globalThis.setTimeout(() => {
+      callCount++;
+    }, 50);
+
+    // Wait 30ms, then refresh (reset the timer)
+    await new Promise((r) => globalThis.setTimeout(r, 30));
```

Comment:
we can use scheduler.wait(30); here instead of this line

Reply:
I'll take it as a nit since it's functionally identical and I'd rather not rerun CI.